src/lib/services/notehub.ts

```diff
@@ -82,30 +84,25 @@ export async function updateDeviceEnvironmentVariablesByPin(
 }
 
 // get events from Airnote project in Notehub
-export async function getEvents(deviceUID: string, timeframe = '30 days') {
+export async function getEvents(deviceUID: string, timeframe = 30) {
   /* this function is fetched on mount with 30 days' worth of data to
 		  populate the CSV download, the AQI average history AND
 		  display today's most current reading as well */
+  const { api_key } = notehubJsClient.authentications;
+  api_key.apiKey = HUB_AUTH_TOKEN;
 
-  const body = {
-    req: 'hub.app.data.query',
-    query: {
-      columns:
-        ".body;.when;lat:(events.value->'best_lat');lon:(events.value->'best_lon');location:(events.value->'best_location')",
-      limit: 10000,
-      order: '.modified',
-      descending: true,
-      where: `.file::text='_air.qo' and .device::text='${deviceUID}' and .modified >= now()-interval '${timeframe}'`
-    }
-  };
+  // convert timeframe into seconds for start and end date
+  const { startDateSecs, endDateSecs } = convertTimeframeToSeconds(timeframe);
 
-  const res = await fetch(
-    `https://api.notefile.net/req?app=${AIRNOTE_PROJECT_UID}`,
-    {
-      method: 'POST',
-      body: JSON.stringify(body)
-    }
-  );
+  const opts = {
+    deviceUID: [deviceUID],
+    sortBy: 'modified',
+    sortOrder: 'desc',
+    files: '_air.qo',
+    startDate: startDateSecs,
+    endDate: endDateSecs,
+    pageSize: 10000
+  };
```

Comment:
You should use `selectFields` to get only the fields that the dashboard uses. Right now the main HTML request to get the dashboard is 1.64MB which is kind of chunky—I bet we could get it down a bunch.

Reply:
Good suggestion. I've made a follow up to do since extracting body-specific values doesn't currently work and the PR fix has yet to be approved in Notehub.